File: coding-ig/desimal2biner/desimal2biner.m
function desimal2biner
    function biner(x)
        _x = x;
        n=0;
        while (x>1)
            x=floor(x/2);
            n+=1;
        end
        for i=n:-1:0
            if (_x >= (2^i))
                digit = floor(_x/(2^i));
                _x -=(digit*(2^i));
                x=_x;
                printf("1 ");
            else
                printf("0 ");
            end
        end
    end

    desimal = input("masukkan desimal = ");
    printf("%d ===>  ",desimal);
    biner(desimal)
end

File: coding-ig/desimal2biner/desimal2biner.py
def biner(x):
    _x = x
    n=0
    while x > 1:
        x//=2
        n +=1
    for i in range(n,-1,-1):
        if _x >= (2**i):
            digit = _x//(2**i)
            _x -= (digit*(2**i))
            x=_x
            print(1, end='')
        else:
            print(0, end='')
        print(" ",end='')

desimal = int(input("masukkan desimal = "))
print(desimal,' ===> ', end='')
biner(desimal)

        
